connectors/sources/generic_database.py

```diff
@@ -172,14 +208,14 @@ async def execute_query(self, query_name, fetch_many=False, **query_kwargs):
                 if fetch_many:
                     # sending back column names only once
                     if yield_once:
-                        if query_kwargs["schema"]:
+                        if kwargs["schema"] is not None:
```

Comment:
Hey @wangch079, We can add an explicit check for it, but I don't think we'll get `KeyError: 'schema'` or `KeyError: 'table'` because we will only need these two when we pass `fetch_many=True`, and we only pass `fetch_many=True` [here](https://github.com/elastic/connectors-python/pull/565/files#diff-e89a6615c27f1f907b620550d9ed4e6681775140019e2be86d9553ed90ae08f2L343) along with schema and table. what are your views on this, should we add a check or not?

Reply:
Noted.

It seems only the query `TABLE_DATA` pass `fetch_many=True`, which is the query to query data from tables, with pagination.

The rest queries are not to query data, but metadata, e.g. ping, get primary keys, get last updated time.

IMO, We shouldn't mix them in one method `execute_query`. But it's not in the scope of this PR.

Reply:
Yes, we can have a follow-up PR for improvements for all SQL-like connectors.